lib/saml_idp/xml_security.rb

```diff
@@ -81,13 +81,15 @@ def signature_algorithm(options)
           algorithm(options[:get_params][:SigAlg])
         else
           ref_elem = REXML::XPath.first(self, "//ds:Reference", {"ds"=>DSIG})
+          return nil unless ref_elem
           algorithm(REXML::XPath.first(ref_elem, "//ds:DigestMethod", {"ds"=>DSIG}))
         end
       end
 
       def fingerprint_cert(cert, options)
         digest_algorithm = signature_algorithm(options)
-        digest_algorithm.hexdigest(cert.to_der)
+        return nil unless digest_algorithm
```

Comment:
oop sorry the suggestion was meant to be an "instead of"
like we remove this line
```suggestion
```

Reply:
Incorporated into https://github.com/18F/saml_idp/pull/80